contracts/cw-core/src/contract.rs

```diff
@@ -181,19 +187,81 @@ pub fn execute_pause(
         .add_attribute("until", until.to_string()))
 }
 
+pub fn execute_admin_msgs(
+    deps: Deps,
+    sender: Addr,
+    msgs: Vec<CosmosMsg<Empty>>,
+) -> Result<Response, ContractError> {
+    let admin = ADMIN.load(deps.storage)?;
+
+    match admin {
+        Some(admin) => {
+            // Check if the sender is the DAO Admin
+            if sender != admin {
+                return Err(ContractError::Unauthorized {});
+            }
+
+            Ok(Response::default()
+                .add_attribute("action", "execute_admin_msgs")
+                .add_messages(msgs))
+        }
+        None => Err(ContractError::NoAdmin {}),
+    }
+}
+
 pub fn execute_proposal_hook(
     deps: Deps,
     sender: Addr,
     msgs: Vec<CosmosMsg<Empty>>,
 ) -> Result<Response, ContractError> {
+    // Check that the message has come from one of the proposal modules
     if !PROPOSAL_MODULES.has(deps.storage, sender) {
         return Err(ContractError::Unauthorized {});
     }
+
     Ok(Response::default()
         .add_attribute("action", "execute_proposal_hook")
         .add_messages(msgs))
 }
 
+pub fn execute_update_admin(
+    deps: DepsMut,
+    sender: Addr,
+    admin: Option<Addr>,
+) -> Result<Response, ContractError> {
+    let current_admin = ADMIN.load(deps.storage)?;
+
+    match current_admin {
+        Some(current_admin) => {
+            // Check sender is the DAO Admin
+            if sender != current_admin {
+                return Err(ContractError::Unauthorized {});
+            }
+
+            // Save the new DAO Admin (which may be set to None)
+            ADMIN.save(deps.storage, &admin)?;
+
+            // Create response with new admin attribute
+            let res;
+            if let Some(admin) = admin {
+                res = Response::default()
+                    .add_attribute("action", "execute_update_admin")
+                    .add_attribute("new_admin", admin.to_string());
+            } else {
+                res = Response::default()
+                    .add_attribute("action", "execute_update_admin")
+                    .add_attribute("new_admin", "none");
+            }
```

Comment:
Might be a cleaner way to do this. Open to suggestions.

Do think it's good to have the new admin as an attribute. @ezekiiel / @Callum-A do we have a standard way of encoding optional attributes? Wrote `"none"` here, but hey it's still a draft.

Reply:
I like to do something like this:
```rust
admin
    .map(|a| a.into_string())
    .unwrap_or_else(|| "None".to_string()),
```
Not terribly simple, but I think it gets the point across reasonably concisely.

Reply:
Great! Nice and clean!